Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/checkins/CheckInRepository.kt

```diff
@@ -29,59 +27,25 @@ class CheckInRepository @Inject constructor(
             .allEntries()
             .map { list -> list.map { it.toCheckIn() } }
 
-    fun addCheckIn(checkIn: CheckIn) {
-        appScope.launch {
-            checkInDao.insert(checkIn.toEntity())
-        }
-    }
+    suspend fun addCheckIn(checkIn: CheckIn) = withContext(NonCancellable) {
+        Timber.d("addCheckIn(checkIn=%s)", checkIn)
+        if (checkIn.id == 0L) throw IllegalArgumentException("ID will be set by DB, ID should be 0!")
 
-    fun updateCheckIn(checkIn: CheckIn) {
-        appScope.launch {
-            checkInDao.update(checkIn.toEntity())
-        }
+        checkInDao.insert(checkIn.toEntity())
     }
 
-    fun clear() {
-        appScope.launch {
-            checkInDao.deleteAll()
-        }
+    suspend fun updateCheckIn(checkInId: Long, update: (CheckIn) -> CheckIn) = withContext(NonCancellable) {
+        Timber.d("updateCheckIn(checkInId=%d, update=%s)", checkInId, update)
```

Comment:
Aren't String templates nicer in Kotlin? 😀

Reply:
If we print plain values like `checkInId`, yes, because there won't be a huge impact, but if we print complex parameters like data classes or callbacks in this case, using formatting args is better for performance, if logging is disabled, as they will not be evaluated `toString()`.

We could now put `checkInId` as template, and update as formatting arg, but.... would this be a blocker for you :slightly_smiling_face: ?

Reply:
Thanks for the explanation. Then its fine for me as it is.